### PR TITLE
[imagetool] Make imagetool responsive

### PR DESCRIPTION
**Pull request by kristofferj**
* Imagetool inside Fieldset
* Nicer previews
* Fixes bug where mousedown is locked when dragging outside window
* Rearrange the listeners to support touch. Also using syntetic events
* Keep cropbox inside boundries when dragging

Some bugs are still present:
* moving hotspot still makes show incorrectly previews while moving
* the code preventing drag crop outside seems to be choppy (should work on the axis that do not goes outside the boundries)

### CODE REVIEW

File: packages/@sanity/imagetool/src/HotspotImage.js [hunk 1]
```diff
@@ -39,6 +40,7 @@ export default class HotspotImage extends React.PureComponent {
   static defaultProps = {
     alignX: 'center',
     alignY: 'center',
+    className: undefined,
```
Comment (bjoerge) [inline packages/@sanity/imagetool/src/HotspotImage.js]: No need to specify undefined as default prop

File: packages/@sanity/imagetool/src/HotspotImage.js [hunk 2]
```diff
@@ -136,7 +138,7 @@ export default class HotspotImage extends React.PureComponent {
       }
     })
     return (
-      <div className={className} style={style} ref={this.setContainerElement}>
+      <div className={className || styles.root} style={style} ref={this.setContainerElement}>
```
Comment (bjoerge) [inline packages/@sanity/imagetool/src/HotspotImage.js]: Shouldn't we merge the classNames? That's the pattern we use elsewhere I think.

File: packages/@sanity/imagetool/src/ImageTool.js [hunk 3]
```diff
@@ -15,6 +16,41 @@ const MARGIN_PX = 8
 const CROP_HANDLE_SIZE = 12
 const HOTSPOT_HANDLE_SIZE = 10
 
+function checkCropBounderies(value, delta) {
```
Comment (bjoerge) [inline packages/@sanity/imagetool/src/ImageTool.js]: Typo bounderies should be boundaries (or just bounds for short)

File: packages/@sanity/imagetool/src/ImageTool.js [hunk 4]
```diff
@@ -15,6 +16,41 @@ const MARGIN_PX = 8
...
+    value.crop.left + delta.left < 0 ||
+    value.crop.right + delta.right < 0 ||
+    value.crop.bottom + delta.bottom < 0
+  ) {
+    return false
+  }
+  return true
+}
+
+function limitToBounderies(value, delta) {
```
Comment (bjoerge) [inline packages/@sanity/imagetool/src/ImageTool.js]: Same typo as mentioned above

File: packages/@sanity/imagetool/src/HotspotImage.js (hunk 1 above)
Comment (kristofferj) [reply packages/@sanity/imagetool/src/HotspotImage.js]: I get a warning from eslint if I don't.

File: packages/@sanity/imagetool/src/HotspotImage.js (hunk 2 above)
Comment (kristofferj) [reply packages/@sanity/imagetool/src/HotspotImage.js]: 👍

File: packages/@sanity/imagetool/src/ImageTool.js (hunk 3 above)
Comment (kristofferj) [reply packages/@sanity/imagetool/src/ImageTool.js]: thx

Comment (bjoerge) [reply packages/@sanity/imagetool/src/ImageTool.js]: Typo still there?

**Pull request merged**